### PR TITLE
Add test showing getting version of root in deployer

### PR DESCRIPTION
**Pull request by AbrilRBS**
Changelog: Omit
Docs: https://github.com/conan-io/docs/pull/XXXX

Some ideas as to how to access the version of the generated package. If this is something we want to document, a similar example might be useful in the docs

### CODE REVIEW

File: conans/test/integration/extensions/test_deployers.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+from conans.test.utils.tools import TestClient
+
+
+def test_deployer_custom_version_output():
+    deployer = textwrap.dedent("""
+    import os
+    def deploy(graph, output_folder, **kwargs):
+        # When running graph info with --requires, the root of the graph is a virtual node, skip it
+        if graph.root.recipe == "Cli":
+            ref_version = str(graph.nodes[1].ref)
```
Comment (memsharded) [inline conans/test/integration/extensions/test_deployers.py:13]: This seems a bit weird, and the access with ``[1]`` probably internal implementation detail that might break in the future.

**Review by memsharded — COMMENTED** [review]
IIUC, the use case is a deployer that works the same for a local ``<path>`` installation than for a ``--requires=mypkg`` installation?
Because it is not only the ``ref`` of the package, but everything changes, like the layout, the folders that need to be managed to do copies or whatever... I am not sure if this wide scope of the deployer is the best approach, but I have no better ideas, lets talk about it.

**Comment by AbrilRBS** [comment]
It working for both ways is just me trying to show that it _can_ be done, but the real use case is to be able to get the ref of the recipe the user is interested in, so that a deploy can be made relative to that name/version. I did it this "general" way so that the linked issue has as much of a staring point base as possible

File: conans/test/integration/extensions/test_deployers.py (hunk 1 above)
Comment (AbrilRBS) [reply conans/test/integration/extensions/test_deployers.py:13]: Yes, you're right. We need to think of a way to make this more handy

**Comment by AbrilRBS** [comment]
Let's maybe document a tiny bit better how the user could find this data if needed, let's move it to the docs

**Pull request closed without merging**